neuromation/cli/job.py

```diff
@@ -877,14 +875,64 @@ def format_fail(job: str, reason: Exception) -> str:
     return job
 
 
+async def _build_volumes(
+    root: Root, input_volumes: Sequence[str], env_dict: Dict[str, str]
+) -> Set[Volume]:
+    VOLUME_SHORTCUTS = {"HOME", "ALL"}
+    detected_shortcuts = VOLUME_SHORTCUTS & set(input_volumes)
+    if len(detected_shortcuts) > 1:
+        msg = ", ".join(f"--volume={shortcut}" for shortcut in detected_shortcuts)
+        raise click.UsageError(f"Cannot use multiple volume shortcuts together: {msg}")
+
+    volumes: Set[Volume] = set()
+    for vol in input_volumes:
+        if vol in VOLUME_SHORTCUTS:
+            if vol == "HOME":
+                volumes.add(
+                    root.client.parse.volume(f"storage://~:{STORAGE_ROOT}/home:rw")
+                )
+                volumes.add(
+                    root.client.parse.volume(
+                        f"storage://neuromation/public:{STORAGE_ROOT}/neuromation:ro"
+                    )
+                )
+            if vol == "ALL":
+                available = await root.client.users.get_acl(
+                    root.username, scheme="storage"
+                )
+
+                env_dict[STORAGE_ROOT_ENV_VAR] = STORAGE_ROOT
+                if not root.quiet:
+                    click.echo(
+                        "Storage root mountpoint will be available as the environment "
+                        f"variable: {STORAGE_ROOT_ENV_VAR}={STORAGE_ROOT}"
+                    )
+                for perm in available:
+                    root_dir = (
+                        perm.uri.host if perm.uri.host != root.username else "home"
+                    )
+                    mountpoint = f"{STORAGE_ROOT}/{root_dir}{perm.uri.path}"
+                    volumes.add(
+                        Volume(
+                            storage_path=str(perm.uri),
+                            container_path=mountpoint,
+                            read_only=perm.action == "read",
+                        )
+                    )
+        else:
+            volumes.add(root.client.parse.volume(vol))
+    return volumes
+
+
 async def upload_and_map_config(root: Root) -> Tuple[str, Volume]:
 
     # store the Neuro CLI config on the storage under some random path
     nmrc_path = URL(root.config_path.expanduser().resolve().as_uri())
     random_nmrc_filename = f"{uuid.uuid4()}-nmrc"
     storage_nmrc_folder = f"storage://{root.username}/nmrc/"
     storage_nmrc_path = URL(f"{storage_nmrc_folder}{random_nmrc_filename}")
-    local_nmrc_folder = "/var/storage/nmrc/"
+    # TODO (artem) if user has dir `storage:nmrc`, `--volume=ALL` will cause a conflict
```

Comment:
Please note that the method `upload_and_map_config` creates directory `/var/storage/nmrc`, so if a user has directory `storage://~/nmrc`, they will be in conflict. Let's maybe change `local_nmrc_folder` to `/var/storage/.nmrc`?

Reply:
the good thing is that we have the corresponding env var for this which allows us to change stuff.

Reply:
not an issue anymore since we decided to use different mountpoints: /var/storage and /var/neuro